sycl/plugins/esimd_emulator/pi_esimd_emulator.cpp

```diff
@@ -957,6 +1016,14 @@ pi_result piMemBufferCreate(pi_context Context, pi_mem_flags Flags, size_t Size,
   }
 
   Status = CmBuf->GetIndex(CmIndex);
+  const std::lock_guard<sycl::detail::SpinLock> Lock{*PiESimdSurfaceMapLock};
+  if (PiESimdSurfaceMap->find((unsigned int)CmIndex->get_data()) !=
```

Comment:
I don't see the rationale for this check `if (PiESimdSurfaceMap->find((unsigned int)CmIndex->get_data())`. Should be an assert at best.

Reply:
Changed to assert.